### PR TITLE
build: Patch API-Extractor to ensure we validate cross-package release tag compatibility

### PR DESCRIPTION
**Pull request by Josmithr**
Mitigation for API-Extractor [issue 4430](https://github.com/microsoft/rushstack/issues/4430).

The original mitigation we had in place was erroneously removed in #19939.

Also fixes the handful of violations that have been introduced since that PR.

### CODE REVIEW

File: packages/common/container-definitions/src/audience.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ import type { IEvent, IEventProvider } from "@fluidframework/core-interfaces";
 import type { IClient } from "@fluidframework/protocol-definitions";
 /**
  * Manages the state and the members for {@link IAudience}
- * @internal
+ * @alpha
```
Comment (Josmithr) [inline packages/common/container-definitions/src/audience.ts:10]: Used by other existing types tagged as `@alpha`

File: packages/tools/devtools/devtools-core/api-report/devtools-core.api.md [hunk 2]
```diff
@@ -89,7 +89,7 @@ export namespace ContainerDevtoolsFeatures {
     }
 }
 
-// @beta
+// @alpha
```
Comment (Josmithr) [inline packages/tools/devtools/devtools-core/api-report/devtools-core.api.md:92]: We don't strictly need to make these `devtools-core` types beta yet, and they reference core framework concepts that are currently `@alpha`. This changes makes them `@alpha` to match.

Comment (Josmithr) [reply packages/tools/devtools/devtools-core/api-report/devtools-core.api.md:92]: Technically breaking, but we have not released this package publicly yet.

File: server/routerlicious/patches/@microsoft__api-extractor@7.42.3.patch [hunk 3]
```diff
@@ -0,0 +1,28 @@
+diff --git a/lib/enhancers/ValidationEnhancer.js b/lib/enhancers/ValidationEnhancer.js
```
Comment (Josmithr) [inline server/routerlicious/patches/@microsoft__api-extractor@7.42.3.patch:1]: `server/routerlicious` requires a separate copy of this to work in their docker environment.

**Review by alexvy86 — APPROVED** [review]

**Pull request merged**